metal/label_model/class_balance.py

```diff
@@ -0,0 +1,136 @@
+from itertools import product
+
+import numpy as np
+import torch
+from torch import nn, optim
+
+
+class ClassBalanceModel(nn.Module):
+    """A model for learning the class balance, P(Y=y), given a subset of LFs
+    which are *conditionally independent*, i.e. \lambda_i \perp \lambda_j | Y,
+    for  i != j.
+
+    Learns the model using a tensor factorization approach.
+
+    Note: This approach can also be used for estimation of the LabelModel, may
+    want to later refactor and expand this class.
+    """
+
+    def __init__(self, k, abstains=True, config=None):
+        super().__init__()
+        self.config = config
+        self.k = k
+        self.abstains = abstains
+        self.k_0 = 0 if self.abstains else 1
```

Comment:
A couple of definitions here for self.k_0 and self.k_lf would be nice, since k can mean a lot of very-similar-but-different-in-important-ways things. Either in the class docstring or here inline.

Reply:
Done!